tests/phpunit/includes/handle-wp-redirect-trait.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Trait to run tests containing `wp_redirect()` and `exit`.
+ * Also privides some assertions for the redirection.
+ */
+trait PLL_Handle_WP_Redirect_Trait {
+	/**
+	 * Whether or not `wp_redirect()` has been called.
+	 *
+	 * @var bool
+	 */
+	protected $has_redirect = false;
+
+	/**
+	 * Stores the last redirection location made with `wp_redirect()`.
+	 *
+	 * @var string
+	 */
+	protected $last_location;
+
+	/**
+	 * Stores the last redirection status made with `wp_redirect()`.
+	 *
+	 * @var int
+	 */
+	protected $last_status;
+
+	/**
+	 * Allows to continue the execution after wp_redirect + exit.
+	 *
+	 * @return void
+	 */
+	protected function handle_wp_redirect() {
+		add_filter(
+			'wp_redirect',
+			function ( $location, $status ) { // phpcs:ignore WordPressVIPMinimum.Hooks.AlwaysReturnInFilter.MissingReturnStatement
+				$this->has_redirect  = true;
+				$this->last_location = $location;
+				$this->last_status   = $status;
+
+				throw new Exception( 'Call to `wp_redirect()`.' );
```

Comment:
I suggest to simplify, just in case we would use this Exception message somewhere else.
```suggestion
				throw new Exception( 'wp_redirect' );
```